server/test/unit/plugins/file/test_distributor.py

```diff
@@ -216,6 +216,55 @@ def test__symlink_units(self):
         expected_symlink_destination = os.path.join(DATA_DIR, self.unit.unit_key['name'])
         self.assertEqual(os.path.realpath(expected_symlink_path), expected_symlink_destination)
 
+    @patch('os.path.exists', return_value=False)
+    @patch('os.makedirs')
+    @patch('os.symlink')
+    def test__symlink_unit_with_subdir(self, mock_symlink, mock_makedirs, mock_exists):
+        """
+        Make sure that if the file being published has to go in a subdirectory, that subdirectory
+        gets created.
+        """
+        distributor = self.create_distributor_with_mocked_api_calls()
+
+        distributor._symlink_unit('/a/b/c/', self.unit, ['d/e/f.txt', ])
+
+        mock_makedirs.assert_called_once_with('/a/b/c/d/e')
+
+    @patch('os.path.exists', return_value=False)
+    @patch('os.makedirs')
+    @patch('os.symlink')
+    def test__symlink_unit_with_existing_subdir(self, mock_symlink, mock_makedirs, mock_exists):
+        """
+        Make sure that if the file being published has to go in a subdirectory, that subdirectory
+        gets created.
+        """
```

Comment:
this docstring probably came from the previous test... subdirectory won't be created in this test